cdap-api/src/main/java/co/cask/cdap/api/Transactional.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api;
+
+import co.cask.cdap.api.data.DatasetContext;
+import co.cask.cdap.api.dataset.Dataset;
+
+/**
+ * An object that executes submitted {@link TxRunnable} tasks. Each task submitted will be executed inside
+ * a transaction.
+ */
+public interface Transactional {
+
+  /**
+   * Execute a set of operations via a {@link TxRunnable} that are committed as a single transaction.
+   * The {@link TxRunnable} can gain access to {@link Dataset} through the {@link DatasetContext} provided
+   * to it.
+   *
+   * @param runnable the runnable to be executed in the transaction
+   */
+  void execute(TxRunnable runnable);
```

Comment:
why does this not throw exceptions? I would think it will throw TransactionFailure/ConflictExceptions.
Also, should this have an option to retry N times in case of conflict?